reconstruction/mltn/src/main/java/org/jlab/service/mltn/MLDCEngine.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
+ * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
+ */
+package org.jlab.service.mltn;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.jlab.clas.reco.ReconstructionEngine;
+import org.jlab.io.base.DataBank;
+import org.jlab.io.base.DataEvent;
+import org.jlab.io.hipo.HipoDataBank;
+import org.jlab.jnp.hipo4.data.Bank;
+import org.jlab.jnp.hipo4.data.CompositeBank;
+
+/**
+ *
+ * @author gavalian
+ */
+public class MLDCEngine extends ReconstructionEngine {
+    
+    
+    public MLDCEngine(){
+        super("MLDC","gavalian","1.0");
+    }
+    
+    @Override
+    public boolean init() {
+        return true;
+    }
+    
+    @Override
+    public boolean processDataEvent(DataEvent de) {
+        
+        List<ByteBuffer> trackBuffers = new ArrayList<>();
+        
+        if(de.hasBank("TimeBasedTrkg::TBTracks")&&de.hasBank("TimeBasedTrkg::TBClusters")){
+            //System.out.println("writing trakcing bank");
+            DataBank   tbt = de.getBank("TimeBasedTrkg::TBTracks");
+            DataBank   tbc = de.getBank("TimeBasedTrkg::TBClusters");
+            ByteBuffer bb = getTracks(tbt,tbc,10);
+            trackBuffers.add(bb);
+            /*DataBank output = de.createBank("MLDC::tracks", bb.array().length);
+            for(int j = 0; j < bb.array().length; j++)
+                output.setByte("bytes", j,bb.array()[j]);
+            de.appendBank(output);*/
+        }
+        
+        if(de.hasBank("TimeBasedTrkg::AITracks")&&de.hasBank("TimeBasedTrkg::AIClusters")){
```

Comment:
Note that the banks with the AI prefix are created only when data-aicv.yaml is used. When data-ai.yaml is used, the tracking banks have the TB prefix even if AI tracking is used.